src/App.tsx

```diff
@@ -1,17 +1,106 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classnames from 'classnames';
 import './App.css';
+import todosFromServer from './api/todos';
+import { TodoList } from './Components/TodoList/TodoLIst';
 
 import users from './api/users';
 
+const preparedTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId)
+      || null,
+  };
+});
+
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState(preparedTodos);
+
+  const onFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (title && userId) {
+      const newTodo = {
+        userId,
+        id: preparedTodos.length + 1,
```

Comment:
should place in functional `setState`. And why you get length from `preparedTodos`.  You add new todos in `todos` but get new id from `preparedTodos` which don't change